tests/unit/pre-build.tap.js

```diff
@@ -0,0 +1,479 @@
+      gypStub.execGyp.yields(null)
+      build('target', false, (err) => {
+        t.error(err)
+        t.ok(gypStub.execGyp.callCount, 2, 'should call execGyp twice')
+        t.same(gypStub.execGyp.args[0][0], ['configure'])
+        t.end()
+      })
+    })
+
+    t.test('should return error if configure fails', (t) => {
+      const expectedErr = new Error('failed to execute cmd')
+      gypStub.execGyp.yields(expectedErr)
+      build('target', false, (err) => {
+        t.same(err, expectedErr)
+        t.end()
+      })
+    })
+
+    t.test('should return error if build fails', (t) => {
+      const expectedErr = new Error('failed to execute cmd')
+      gypStub.execGyp.onCall(0).yields(null)
+      gypStub.execGyp.onCall(1).yields(expectedErr)
+      build('target', false, (err) => {
+        t.same(err, expectedErr)
+        t.end()
+      })
+    })
+  })
+
+  t.test('should move build accordingly', (t) => {
+    sinon.stub(fs, 'rename')
+    t.teardown(() => {
+      fs.rename.restore()
+    })
+
+    preBuild.moveBuild('target', (err) => {
+      t.error(err)
+      t.equal(fs.rename.callCount, 1)
+      t.end()
+    })
+
+    // call cb manually to end test
+    fs.rename.args[0][2]()
+  })
+
+  t.test('download', (t) => {
+    t.autoend()
+    let sandbox
+
+    t.before(() => {
+      nock.disableNetConnect()
+    })
+
+    t.beforeEach(() => {
+      sandbox = sinon.createSandbox()
+    })
+
+    t.afterEach(() => {
+      sandbox.restore()
+      delete process.env.NR_NATIVE_METRICS_PROXY_HOST
+      delete process.env.NR_NATIVE_METRICS_DOWNLOAD_HOST
+    })
+
+    t.test('should download and unzip file accordingly', (t) => {
+      const expectedData = Buffer.from('testing', 'utf-8')
+      const file = zlib.gzipSync(expectedData)
+      nock('https://download.newrelic.com/').get(/.*/).reply(200, file)
+
+      preBuild.download('test', (err, data) => {
+        t.error(err)
+        t.same(data, expectedData)
+        t.end()
+      })
+    })
+
+    t.test('should return error if 404 occurs', (t) => {
+      nock('https://download.newrelic.com/').get(/.*/).reply(404)
+
+      preBuild.download('test', (err) => {
+        t.equal(err.message, 'No pre-built artifacts for your OS/architecture.')
+        t.end()
+      })
+    })
+
+    t.test('should return failed to download error if response is not 200 nor 404', (t) => {
+      nock('https://download.newrelic.com/').get(/.*/).reply(500)
+
+      preBuild.download('test', (err) => {
+        t.match(err.message, /Failed to download.*code 500/)
+        t.end()
+      })
+    })
+
+    t.test('should fail if it cannot unzip', (t) => {
+      const expectedData = Buffer.from('testing', 'utf-8')
+      nock('https://download.newrelic.com/').get(/.*/).reply(200, expectedData)
+
+      preBuild.download('test', (err) => {
+        t.match(err.message, /Failed to unzip.*/)
+        t.end()
+      })
+    })
+
+    /* TODO: not sure how to test when res is a stream
```

Comment:
Do we still want to save this for later, or remove?

Reply:
i wanted to keep i can provide more context or i can remove, thoughts?

Reply:
I'm cool with keeping if we plan on actually enabling the test at some point, otherwise I'd say remove

Reply:
i just removed it